app/webpacker/components/Panel/pages/HelpfulQueriesPage/index.jsx

```diff
@@ -0,0 +1,212 @@
+import React, { useMemo } from 'react';
+import {
+  Form, Header, Label, Loader, Message, Segment, Tab, Table,
+} from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
+import { IdWcaSearch } from '../../../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
+import { fetchJsonOrError } from '../../../../lib/requests/fetchWithAuthenticityToken';
+import { viewUrls, competitionUrl } from '../../../../lib/requests/routes.js.erb';
+import useInputState from '../../../../lib/hooks/useInputState';
+
+const hasWcaId = (val) => val?.trim()?.length > 0;
+
+const statusColor = (s) => {
+  switch ((s?.toLowerCase() || '')) {
```

Comment:
Tiny nitpick: I still don't see why the `|| ''` part is necessary here. If for some reason `s` is null or undefined, then `s?.toLowerCase()` will become null and that `null` value will trigger the `default` branch the same way that an empty string literal `''` would.